heartbeat.go

```diff
@@ -60,24 +61,20 @@ func (k *Kite) kontrolFunc(fn kontrolFunc) error {
 // use it during app initializations. After the registration a reconnect is
 // automatically handled inside the RegisterHTTP method.
 func (k *Kite) RegisterHTTPForever(kiteURL *url.URL) {
-	interval := time.NewTicker(kontrolRetryDuration)
-	defer interval.Stop()
-
-	_, err := k.RegisterHTTP(kiteURL)
-	if err == nil {
-		return
-	}
-
-	for _ = range interval.C {
+	register := func() error {
 		_, err := k.RegisterHTTP(kiteURL)
-		if err == nil {
-			return
+		if err != nil {
+			k.Log.Error("Cannot register to Kontrol: %s Will retry after %d seconds",
+				err,
+				k.httpRegisterBackOff.NextBackOff()/time.Second)
+			return err
 		}
 
-		k.Log.Error("Cannot register to Kontrol: %s Will retry after %d seconds",
-			err, kontrolRetryDuration/time.Second)
-
+		return nil
 	}
+
+	// this will retry register forever
+	backoff.Retry(register, k.httpRegisterBackOff)
```

Comment:
Sure!

For what it's worth though i believe all errors are already logged. [Retry](https://github.com/cenkalti/backoff/blob/master/retry.go#L26) only returns errors from the operation, and in this case the operation itself _(the anonymous func `register)_ logs all errors [on line 66](https://github.com/koding/kite/pull/127/files#diff-c4c61e16bcef8f4df86f5aa1902e4803R66). Probably best to log the returned (final) error regardless :)